### ISSUE TITLE
Some inserts producing invalid INSERT errors

### ISSUE THREAD

**Issue by scottlimmer** [timescale/timescaledb#1298]
**Relevant system information:**
 - OS: Ubuntu 18.04.2 LTS
 - PostgreSQL version (output of `postgres --version`): postgres (PostgreSQL) 10.8 (Ubuntu 10.8-1.pgdg18.04+1)
 - TimescaleDB version (output of `\dx` in `psql`): 1.3.1
   - TimescaleDB has been updated in place multiple times 
 - Installation method: apt install

**Describe the bug**
Certain inserts are producing the following error: E.g.:
```
INSERT INTO "OBSERVATIONS" ( "DATASTREAM_ID", "PHENOMENON_TIME_END", "PHENOMENON_TIME_START", "RESULT_TIME", "RESULT_TYPE", "RESULT_STRING", "RESULT_NUMBER", "RESULT_BOOLEAN", "RESULT_JSON", "PARAMETERS", "FEATURE_ID" )
VALUES ( '6', '2018-12-11 15:30:00+11', '2018-12-11 15:30:00+11', '2018-12-11 15:30:00+11', '0', '0.7481988', '0.748198800000000053', NULL, NULL, NULL, '63' )
```
``` 
ERROR:  invalid INSERT on the root table of hypertable "_hyper_1_496_chunk"
HINT:  Make sure the TimescaleDB extension has been preloaded.
```
However the following with only the "RESULT_TIME" value changed, inserts correctly:
```
INSERT INTO "OBSERVATIONS" ( "DATASTREAM_ID", "PHENOMENON_TIME_END", "PHENOMENON_TIME_START", "RESULT_TIME", "RESULT_TYPE", "RESULT_STRING", "RESULT_NUMBER", "RESULT_BOOLEAN", "RESULT_JSON", "PARAMETERS", "FEATURE_ID" )
VALUES ( '6', '2018-12-11 15:30:00+11', '2018-12-11 15:30:00+11', '2019-12-11 15:30:00+11', '0', '0.7481988', '0.748198800000000053', NULL, NULL, NULL, '63' )
```

~~As far as I've discovered, this only appears to be happening on chunk 496. This makes me suspect some form of corruption.~~ This is happening on more than one chunk, but not all. I've tried `vacuum full` and `reindex table "OBSERVATIONS"` to no effect.

See structure for offending hypertable
[OBSERVATIONS-schema.txt](https://github.com/timescale/timescaledb/files/3289352/OBSERVATIONS-schema.txt)

**Comment by dianasaur323** [comment]
@cevian do you mind assigning someone to this? Ante mentioned that we would want someone who works on the core side to take a deeper look.

**Comment by scottlimmer** [comment]
@mfreed Timescale is definitely loaded. This error is occurring as part of a larger data migration from an old non-TSDB schema; so not a restore. The majority of insert statements complete successfully, it appears to be only chunk 496 that produces errors.
`\dx` output below, I've also attached the full TimescaleDB meta dump

```
sensorthings=# \dx
                                       List of installed extensions
    Name     | Version |   Schema   |                             Description
-------------+---------+------------+---------------------------------------------------------------------
 plpgsql     | 1.0     | pg_catalog | PL/pgSQL procedural language
 postgis     | 2.4.3   | public     | PostGIS geometry, geography, and raster spatial types and functions
 timescaledb | 1.3.1   | public     | Enables scalable inserts and complex queries for time-series data
(3 rows)
```

[timescaledb_meta_dump.txt](https://github.com/timescale/timescaledb/files/3303398/timescaledb_meta_dump.txt)

**Comment by gayyappan** [comment]
@scottlimmer Thank you for uploading the files. I see 1) insert_blocker trigger on _hyper_1_496_chunk -> this blocks all the inserts on this chunk and 2) CLUSTER INDEX on this chunk but not the other one.

ALTER TABLE "_timescaledb_internal"."_hyper_1_496_chunk" 
  OWNER TO "sensorthings_user",
  CLUSTER ON "_hyper_1_496_chunk_OBSERVATIONS_DATASTREAM_ID_RESULT_TIME_idx";

When did you start the cluster?  ( I don't see this at the table level).   Are you able to select from this chunk? Has the CLUSTER operation completed?

**Comment by scottlimmer** [comment]
@gayyappan As far as I'm aware I've never started a cluster, certainly never intentionally so. Is there a way to revert this back to single node?

**Comment by mfreed** [comment]
@scottlimmer Sorry, confusing naming:  CLUSTER is a function that reorders the layout of a table according to an index, it has nothing to do with distributed operation.

https://www.postgresql.org/docs/current/sql-cluster.html

**Comment by scottlimmer** [comment]
@mfreed @gayyappan Ah yes, I did manually try to reorder chunk 496, but the issue existed before this so this isn't relevant the CLUSTER option isn't relevant. Sorry for the confusion. I've reverted this and attached an updated DDL for chunk 496.

The not working chunks still have trigger `ts_insert_blocker` and time constraint with my local timezone different to the working chunks.

**Comment by gayyappan** [comment]
@scottlimmer  We need to figure out the sequence of events that added "ts_insert_blocker" to the chunks.  1) What was the update between the chunks that work and chunks that don't?  Could you describe this in a bit more detail. You mention that you had migrated data.  a) did you migrate from a regular postgres tables to the Timescaledb on the same database? b) What were the timescaledb version upgrades? c) What is the migration process like -> are you inserting data into these tables from the old tables after you created an empty hypertable or something else? d) I see continuous aggs on a table. Is the view on the affected table?  When did you create that?

**Comment by gayyappan** [comment]
@scottlimmer Which tables have ts_insert_blocker? 
``` sql
select * from information_schema.triggers where trigger_name like 'ts_insert_blocker';
```
Are these the only tables with different time zone constraints or everything before the update has different timezone and everything after a different one?

**Comment by scottlimmer** [comment]
> @scottlimmer We need to figure out the sequence of events that added "ts_insert_blocker" to the chunks. 1) What was the update between the chunks that work and chunks that don't? Could you describe this in a bit more detail. You mention that you had migrated data. a) did you migrate from a regular postgres tables to the Timescaledb on the same database? b) What were the timescaledb version upgrades? c) What is the migration process like -> are you inserting data into these tables from the old tables after you created an empty hypertable or something else? d) I see continuous aggs on a table. Is the view on the affected table? When did you create that?

* Installed TimescaleDB 1.2.2. 
* Initial schema was created via [FROST-Server](https://github.com/FraunhoferIOSB/FROST-Server) (a SensorThings API implementation). 
* Created hypertable on an empty "OBSERVATIONS" table. 
* Removed necessary indexes/triggers from schema 
* Initial load of historic data was done via PSQL from a binary dump on a testing database. 
* Reapplied indexes/triggers after data load.
* Updated TSDB to 1.3.0
* Attempted to load data via FROST-Server API. Only newly created chunks working
* Updated TSDB to 1.3.1
* Created continuous aggregate on "OBSERVATIONS".

**Comment by scottlimmer** [comment]
> @scottlimmer Which tables have ts_insert_blocker?
> 
> ```sql
> select * from information_schema.triggers where trigger_name like 'ts_insert_blocker';
> ```
> 
> Are these the only chunk tables which show different time zone constraints (+00 versus +10) or do all the chunks before the update have a different timezone and everything after a different one? Did the sessions (where you executed the queries for the the chunks and dumped _hyper_1****chunk.txt) have different timezones by any chance?

See attached ts_insert_blocker list
[ts_insert_blocker.txt](https://github.com/timescale/timescaledb/files/3312331/ts_insert_blocker.txt)

I've done some more investigation, it appears the timezone issue might also be irrelevant. For some reason Navicat generates DDLs for the newer chunks with contraint timestamps in +10. However when I select directly from information_schema.check_constraints all timestamps are +00.

**Comment by gayyappan** [comment]
@scottlimmer Could you give me the output for the following from psql (hard to parse the insert stmts from ts_insert_blocker.txt)? Thsi will idnetify just the chunk tables that have the "ts_insert_blocker"
``` sql
select ch.* , pg_trigger.* from _timescaledb_catalog.chunk ch, pg_class , pg_trigger where pg_class.relname = ch.table_name and pg_trigger.tgrelid = pg_class.oid and tgname like 'ts_insert_blocker';
```
1.Based on the output above, could you classify the affected chunks? Are these chunks before update (all the chunks before update/ all hypertables etc.)? 
2.What triggers and indexes were dropped before the load and how were they reapplied  before any of the TimescaleDb updates.

**Comment by scottlimmer** [comment]
[ts_blocker_list.txt](https://github.com/timescale/timescaledb/files/3318649/ts_blocker_list.txt)

All chunks in that output are before updates. 
All indexes/triggers except the one created by create_hypertable('"OBSERVATIONS"', 'RESULT_TIME') were dropped.

**Comment by gayyappan** [comment]
@scottlimmer My conjecture is that either reapplying trigger inadvertently added triggers to the tables in the inheritance hierarchy for the hypertable chunks or something failed during the update. Do you know if the chunks were working before the update? Do you have a backup from before the update to check the status of the tables?

To get the chunks working again, please do the following:
``` sql
drop trigger ts_insert_blocker on _timescaledb_internal.<...._chunk>;  for the chunks identified by ts_blocker_list.txt
```

**Comment by scottlimmer** [comment]
@gayyappan Unfortunately this hasn't transitioned to production (i.e.: no backups). It's possible triggers have been reapplied incorrectly. I don't know if chunks were working before the updates, no attempts were made to update data after initial load/before update. 

Thanks for taking the time to investigate. Much appreciated.

**Comment by gayyappan** [comment]
@scottlimmer  The ts_insert_blocker trigger  is defined only on the parent hypertable. None of the chunk tables _timescaledb_internal._hyper_xxxx_chunk should have this trigger.  So please  remove the trigger only from the chunk tables.

**Comment by gayyappan** [comment]
@scottlimmer Closing this issue.

**Comment by jitesh9** [comment]
Hello,
I am also getting the following error:
ERROR:  invalid INSERT on the root table of hypertable "_hyper_2_19702_chunk"
HINT:  Make sure the TimescaleDB extension has been preloaded.
SQL state: 0A000

What did I change that caused this error?
The disk where Postgresql is installed was nearly full.
I renamed the hypertable from tablename to tablename_old.  Then I created a new "table" on a new tablespace which was created on the new disk.
Then I created table as a new hypertable.

When I run this query:
select * from information_schema.triggers where trigger_name like 'ts_insert_blocker';
I do see both "table" and "table_old" in my result of the query.

Please guide me on how I can insert data in my new "table".

**Comment by nathanl** [comment]
I have this issue on a local machine where I've restored a dump of production data. In general, restoring from dumps production seems tricky; I get a lot of warnings. It would be nice to have timescale-friendly tools for dumping and loading data.

**Comment by anthonyma94** [comment]
I'm having this issue in 2024, on pg15, timescaledb v2.11.2, and timescaledb_toolkit v1.17.0

**Comment by Kies8** [comment]
> I'm having this issue in 2024, on pg15, timescaledb v2.11.2, and timescaledb_toolkit v1.17.0

the latest version is 2.16.1
